src/discovery/Discovery.ts

```diff
@@ -175,299 +205,291 @@ class Discovery {
-                  vertexChainData[claimId] = claimsUtils.decodeClaim(
-                    vertexChainDataEncoded[claimId],
-                  );
-                }
-                // Otherwise, request the verified chain data from the node
-              } else {
-                try {
-                  vertexChainData = await this.nodeManager.requestChainData(
-                    nodeId,
-                  );
-                } catch (e) {
-                  this.visitedVertices.add(vertex);
-                  await this.removeKeyFromDiscoveryQueue(vertexId);
+  protected processVertex(
+    vertex: GestaltKey,
+    ctx?: ContextTimed,
+  ): PromiseCancellable<void>;
+  @TimedCancellable(true, 20000)
+  protected async processVertex(
+    vertex: GestaltKey,
+    @context ctx: ContextTimed,
+  ): Promise<void> {
+    this.logger.debug(`Processing vertex: ${vertex}`);
+    const vertexGId = gestaltsUtils.ungestaltKey(vertex);
+    switch (vertexGId.type) {
+      case 'node':
+        {
+          // The sigChain data of the vertex (containing all cryptolinks)
+          let vertexChainData: ChainData = {};
+          // If the vertex we've found is our own node, we simply get our own chain
+          const nodeId = nodesUtils.decodeNodeId(vertexGId.nodeId)!;
+          if (nodeId.equals(this.keyManager.getNodeId())) {
+            const vertexChainDataEncoded = await this.sigchain.getChainData();
+            // Decode all our claims - no need to verify (on our own sigChain)
+            for (const c in vertexChainDataEncoded) {
+              const claimId = c as ClaimIdEncoded;
+              vertexChainData[claimId] = claimsUtils.decodeClaim(
+                vertexChainDataEncoded[claimId],
+              );
+            }
+            // Otherwise, request the verified chain data from the node
+          } else {
+            try {
+              vertexChainData = await this.nodeManager.requestChainData(nodeId);
+            } catch (e) {
+              this.visitedVertices.add(vertex);
+              this.logger.error(
+                `Failed to discover ${vertexGId.nodeId} - ${e.toString()}`,
+              );
+              return;
+            }
+          }
+          // TODO: for now, the chain data is treated as a 'disjoint' set of
+          //  cryptolink claims from a node to another node/identity
+          //  That is, we have no notion of revocations, or multiple claims to
+          //  the same node/identity. Thus, we simply iterate over this chain
+          //  of cryptolinks.
+          // Now have the NodeInfo of this vertex
+          const vertexNodeInfo: NodeInfo = {
+            id: nodesUtils.encodeNodeId(nodeId),
+            chain: vertexChainData,
+          };
+          // Iterate over each of the claims in the chain (already verified)
+          // TODO: because we're iterating over keys in a record, I don't believe
+          //  that this will iterate in lexicographical order of keys. For now,
```

Comment:
The comments still seem relevant but I'm not sure why they're TODOs because they don't describe something that needs to be done. They're just descriptive. I can make them normal comments?

Reply:
It says `I don't believe that this will iterate in lexicographical order of keys. For now,`. What does this mean? Can we make confirm what the proper behaviour is and rewrite the comment.

Reply:
I'm going to leave these as is. The gist of what they're saying is that if we had revocation then the order here would matter. But since we don't support that right now it doesn't matter what order we iterate over the claims.  If we do want to update it then I'll leave it for the next stage of discovery refactoring.

Reply:
Do you mean sigchain revocation of claims?

Reply:
What order are we iterating claims atm? Wouldn't we always iterate from most recent to most oldest?

Reply:
This comment is referenced in #327. I'll update the comment to make it less weird.